source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation_data.py

```diff
@@ -269,6 +272,35 @@ def root_state_w(self):
             self._root_state_w.timestamp = self._sim_timestamp
         return self._root_state_w.data
 
+    @property
+    def root_state_link_w(self):
+        """Root state ``[pos, quat, lin_vel, ang_vel]`` in simulation world frame. Shape is (num_instances, 13).
+
+        The position, quaternion, and linear/angular velocity are of the articulation root's actor frame relative to the
+        world.
+        """
+        state = self.root_state_w.clone()
+        quat = state[:, 3:7]
+        # adjust linear velocity to link
+        state[:, 7:10] += torch.linalg.cross(
+            state[:, 10:13], math_utils.quat_rotate(quat, -self._com_pos_b[:, 0, :]), dim=-1
+        )
+        return state
```

Comment:
Makes sense to make this also a lazy buffer to not recompute everytime?

Reply:
Oh yeah. I naively thought that by pulling from `self.root_state_w` instead of `self._root_state_w` it would be lazy, but the additional calculations over top are not lazy. So yeah will do.